dags/multipod/maxtext_end_to_end.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A DAG to run end-to-end MaxText tests."""
+
+
+import datetime
+from airflow import models
+from dags import composer_env
+from dags.vm_resource import TpuVersion, Zone
+from dags.multipod.configs import maxtext_gce_config
+from dags.multipod.configs.common import SetupMode
+
+
+# Run once a day at 4 am UTC (8 pm PST)
+SCHEDULED_TIME = "0 4 * * *" if composer_env.is_prod_env() else None
+
+
+with models.DAG(
+    dag_id="maxtext_end_to_end",
+    schedule=SCHEDULED_TIME,
+    tags=["multipod_team", "maxtext", "stable", "nightly"],
+    start_date=datetime.datetime(2024, 1, 19),
+    catchup=False,
+) as dag:
+  test_name_prefix = "maxtext"
+  test_models = ["llama2", "mistral", "gamma"]
+  test_modes = [SetupMode.STABLE, SetupMode.NIGHTLY]
+
+  for model in test_models:
+    for mode in test_modes:
```

Comment:
Could we add a concurrency limit or some dependencies between the tests? I'm always paranoid about capacity failures.

Reply:
Discussed offline. It seems not a major concern for now, as we have ~160 cores quota for v4 TPUs. But we may consider to add concurrency limit in the future when the num of tests increase.